src/Engine/ConfigurationState.php

```diff
@@ -0,0 +1,91 @@
+<?php
+
+/**
+ * @file
+ * Contains \Drupal\rules\Engine\ConfigurationState.
+ */
+
+namespace Drupal\rules\Engine;
+
+use Drupal\Core\Config\Entity\ConfigEntityInterface;
+use Drupal\Core\Language\LanguageInterface;
+use Drupal\Core\TypedData\DataDefinitionInterface;
+use Drupal\rules\Entity\ReactionRuleConfig;
+
+/**
+ * The state used during configuration time holding data definitions.
+ */
+class ConfigurationState implements ConfigurationStateInterface {
```

Comment:
Maybe "PreExecutionState" would be clearer? Else, it's hard to grasp what the state is used for.